src/yunohost/app.py

```diff
@@ -65,66 +69,122 @@ def app_listlists():
     """
     List fetched lists
 
-
     """
-    list_list = []
-    try:
-        for filename in os.listdir(REPO_PATH):
-            if '.json' in filename:
-                list_list.append(filename[:len(filename)-5])
-    except OSError:
-        raise MoulinetteError(1, m18n.n('no_appslist_found'))
 
-    return { 'lists' : list_list }
+    # Migrate applist system if needed
+    # XXX move to a migration when those are implemented
+    if _using_legacy_applist_system():
+        _migrate_applist_system()
+
+    # Get the list
+    applist_list = _read_applist_list()
+
+    return applist_list
 
 
 def app_fetchlist(url=None, name=None):
     """
-    Fetch application list from app server
+    Fetch application list(s) from app server. By default, fetch all lists.
 
     Keyword argument:
-        name -- Name of the list (default yunohost)
-        url -- URL of remote JSON list (default https://app.yunohost.org/official.json)
-
+        name -- Name of the list
+        url -- URL of remote JSON list
     """
-    # Create app path if not exists
+    # If needed, create folder where actual applists are stored
     if not os.path.exists(REPO_PATH):
         os.makedirs(REPO_PATH)
 
-    if url is None:
-        url = 'https://app.yunohost.org/official.json'
-        name = 'yunohost'
-    elif name is None:
-        raise MoulinetteError(errno.EINVAL,
-                              m18n.n('custom_appslist_name_required'))
+    # Migrate applist system if needed
+    # XXX move that to a migration once they are finished
+    if _using_legacy_applist_system():
+        _migrate_applist_system()
+
+    # Read the list of applist...
+    applists = _read_applist_list()
+
+    # Determine the list of applist to be fetched
+    applists_to_be_fetched = []
+
+    # If a url and and a name is given, try to register new list,
+    # the fetch only this list
+    if url is not None:
+        if name:
+            _register_new_applist(url, name)
```

Comment:
I was trying to implement this, and realized that it might not be such a good idea : during the postinstall, we add the official.json list. If we implement this and someone does an offline postinstall, then the official.json won't be added, which we don't want to happen. Same goes if yunohost.org (or whatever list server) is down when trying to add a list which the user knows really exists.

Or we can have an option to disable the check... But I think people should (by default) be able to add a list independently from the internet connection and the availability of the remote server. If the list really doesn't exist, the renew cron job will anyway send an email to root when after attempting to fetch the list, and he'll end up removing the list.